test.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python3
+
+import logging
+import os
+import shutil
+import subprocess
+import sys
+
+# constants
+TEST_FILE = "/data/test.txt"
+RED = "\033[31m"
+GREEN = "\033[32m"
+MAGENTA = "\033[95m"
+RESET = "\033[0m"
+
+LOGFILE = 'test.log'
+
+logger = logging.getLogger(__file__)
+logger.setLevel(logging.DEBUG)
+
+
+class OsmaxxTestSuite:
+
+    def main(self):
+        # This function will be called when running this script
+        if os.path.samefile('docker-compose.yml', 'compose-development.yml'):
+            self.run_development_tests()
+        else:
+            self.run_production_tests()
+
+        # FIXME: currently only works on development settings
+        if os.path.samefile('docker-compose.yml', 'compose-development.yml') and os.environ.get('RUN_E2E') == 'true':
+            self.run_e2e_tests()
+
+    def run_e2e_tests(self):
+        with TmpVirtualEnv() as tmp_venv:
```

Comment:
In this local context the variable name is o.k. but in other context nobody would understand what ´tmp_venv´ is.